app/tools/research.py

```
@@ -0,0 +1,392 @@
+            headers=headers,
+            json=body,
+            stream=True,
+            timeout=(30, timeout_seconds),  # (connect, read)
+        )
+    except requests.exceptions.RequestException as e:
+        return {"error": f"failed to reach research endpoint: {e}"}
+
+    if resp.status_code != 200:
+        try:
+            err_body = resp.text[:500]
+        except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)